In [77]:
import os
import pandas as pd

def get_csv_filenames_with_min_rows(directory, min_rows,max_rows):
    csv_files = []
    for filename in os.listdir(directory):
        if filename.endswith('.csv'):
            file_path = os.path.join(directory, filename)
            try:
                # Read the CSV file with pandas
                df = pd.read_csv(file_path)
                # Check if the number of rows is greater than or equal to the minimum specified
                if (len(df) >= min_rows) and (len(df)<=max_rows):
                    csv_files.append(filename)
            except Exception as e:
                print(f"Error reading file {filename}: {e}")
    return csv_files

# Directory containing CSV files
directory_path = 'animation-songwen-final'
# Minimum number of rows required
min_rows = 194
max_rows = 1000
csv_files_with_min_rows = get_csv_filenames_with_min_rows(directory_path, min_rows,max_rows)
print("CSV files with 194 or more rows:")
print(len(csv_files_with_min_rows))
# for csv in csv_files_with_min_rows:
#     print(csv)


CSV files with 194 or more rows:
184


In [78]:
success_list=[]
for csv in csv_files_with_min_rows:
    temp=pd.read_csv('animation-songwen-final/'+csv)
    temp=temp[temp['trial_type']=='survey-text']
    success_list.append(list(temp['response'])[0])
len(list(set(success_list)))

183

305.csv: 5f4520ebb012170878012efc # this person try again
390.csv: 6032a88a2b8e882a37cc2766 # this person not

In [79]:
'{"Q0":"5f4520ebb012170878012efc"}' in success_list
'{"Q0":"6032a88a2b8e882a37cc2766"}' in success_list


False

In [80]:
merge_data=pd.read_csv('animation-songwen-final.csv')
merge_data.head(5)

rt                                           stimulus response  \
0  7553.0  <p>Due to compatibility issues, please <b>do n...        0   
1     NaN                                                NaN      NaN   
2     NaN                                                NaN      NaN   
3  6420.0  <p>Due to compatibility issues, please <b>do n...        0   
4     NaN                                                NaN      NaN   

             trial_type  trial_index  time_elapsed internal_node_id  run_id  \
0  html-button-response            0          7563          0.0-0.0     155   
1         browser-check            1          8554          0.0-1.0     155   
2         browser-check            2         19370          0.0-2.0     155   
3  html-button-response            0          6428          0.0-0.0     153   
4         browser-check            1          7417          0.0-1.0     153   

   condition          recorded_at  ... item_height_mm item_width_px px2mm  \
0          1  2024-02-29 04:45:42  ...            NaN           NaN   NaN   
1          1  2024-02-29 04:45:42  ...            NaN           NaN   NaN   
2          1  2024-02-29 04:45:42  ...            NaN           NaN   NaN   
3          1  2024-02-29 04:45:38  ...            NaN           NaN   NaN   
4          1  2024-02-29 04:45:38  ...            NaN           NaN   NaN   

  scale_factor correct question_order init_locations moves final_locations  \
0          NaN     NaN            NaN            NaN   NaN             NaN   
1          NaN     NaN            NaN            NaN   NaN             NaN   
2          NaN     NaN            NaN            NaN   NaN             NaN   
3          NaN     NaN            NaN            NaN   NaN             NaN   
4          NaN     NaN            NaN            NaN   NaN             NaN   

  accuracy  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  

[5 rows x 42 columns]

In [81]:
prolific=merge_data[merge_data['trial_type']=='survey-text']
prolific.head()

rt stimulus                           response   trial_type  \
378   5891.0      NaN  {"Q0":"59b94b6211212f0001cd118a"}  survey-text   
391   8436.0      NaN  {"Q0":"65ab04a12a188d698a4df098"}  survey-text   
588   6482.0      NaN  {"Q0":"5ee454ad40c72e152991826e"}  survey-text   
787  36437.0      NaN  {"Q0":"650c46a95d672d7602328705"}  survey-text   
988   7302.0      NaN  {"Q0":"613b45335ef9d60c2f15bb86"}  survey-text   

     trial_index  time_elapsed internal_node_id  run_id  condition  \
378            6         35396          0.0-6.0     110          1   
391            6         87433          0.0-6.0     103          1   
588            6        133165          0.0-6.0      12          1   
787            6         76385          0.0-6.0      11          1   
988            6        172985          0.0-6.0     104          1   

             recorded_at  ... item_height_mm item_width_px px2mm scale_factor  \
378  2024-02-29 04:45:03  ...            NaN           NaN   NaN          NaN   
391  2024-02-29 04:44:45  ...            NaN           NaN   NaN          NaN   
588  2024-02-29 02:28:46  ...            NaN           NaN   NaN          NaN   
787  2024-02-29 02:28:17  ...            NaN           NaN   NaN          NaN   
988  2024-02-29 04:44:48  ...            NaN           NaN   NaN          NaN   

    correct question_order init_locations moves final_locations accuracy  
378     NaN            NaN            NaN   NaN             NaN      NaN  
391     NaN            NaN            NaN   NaN             NaN      NaN  
588     NaN            NaN            NaN   NaN             NaN      NaN  
787     NaN            NaN            NaN   NaN             NaN      NaN  
988     NaN            NaN            NaN   NaN             NaN      NaN  

[5 rows x 42 columns]

In [82]:
len(prolific.response.tolist())

202

In [83]:
all_people=list(set(prolific.response.tolist()))
print(len(all_people))
for people in all_people:
    if people not in success_list:
        print(people)

189
{"Q0":"65257c1a7780ceee7f6ef9bd"}
{"Q0":"6032a88a2b8e882a37cc2766"}
{"Q0":"59b94b6211212f0001cd118a"}
{"Q0":"5a520774eedc3200014206b4"}
{"Q0":"65959b14d677879036710db0"}
{"Q0":"60fd125837662fc5b446b2b8"}


Among 189 collected runs, 183 of them are valid. 1 of them are almost valid, missing the ranking and demographics.

6032a88a2b8e882a37cc2766 finish almost everything. Let's fund this person.

The following 5 ID didn't finish:
65257c1a7780ceee7f6ef9bd;
59b94b6211212f0001cd118a;
5a520774eedc3200014206b4;
65959b14d677879036710db0;
60fd125837662fc5b446b2b8;